### PR TITLE
Improve minifier printing to be more chatty when it makes sense

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100546
* #100519
* __->__ #100486
* #100447
* #100472
* #100471
* #100467

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_dynamo/repro/after_aot.py [hunk 1]
```diff
@@ -559,14 +559,16 @@ def isolate_fails(
...
-    return False
+    stdout.seek(0)
+    stderr.seek(0)
+    print(
+        textwrap.indent(stdout.read().decode("utf-8"), prefix=">>  "), file=sys.stdout
+    )
+    print(
+        textwrap.indent(stderr.read().decode("utf-8"), prefix=">>  "), file=sys.stderr
+    )
+    # print(f"Isolated test failed - {file_name}")
```
Comment (voznesenskym) [inline torch/_dynamo/repro/after_aot.py:570]: Why comment?

File: torch/_functorch/fx_minifier.py [hunk 2]
```diff
@@ -95,13 +96,17 @@ def graph_fails(graph, inps):
...
         raise RuntimeError("Input graph did not fail the tester")
-    print(f"Started off with {cur_size} nodes")
+    print(f"Started off with {cur_size} nodes", file=sys.stderr)
 
     def _register_strategy(strategy: Callable, name: str):
         @wraps(strategy)
         def new_func(old_state: ReproState, granularity=1):
-            print()
-            print(f"Strategy: {name} (G: {granularity}) ({len(old_state.graph.nodes)} nodes, {len(old_state.inps)} inputs)")
+            print(file=sys.stderr)
```
Comment (voznesenskym) [inline torch/_functorch/fx_minifier.py:104]: nit: If you're gonna jam all the prints towards `file=sys.stderr` consider doing something like

print_outstream = sys.stderr

And then if you change your mind, you only need to do flip flop it once

**Review by voznesenskym — APPROVED** [review]
lgtm

File: torch/_dynamo/repro/after_aot.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/repro/after_aot.py:570]: that's what the old thing did

File: torch/_functorch/fx_minifier.py (hunk 2 above)
Comment (ezyang) [reply torch/_functorch/fx_minifier.py:104]: it seems right for this to go to stderr though, it should always be line buffered and it's not intended to be parseable

**Pull request closed without merging**